Cannot build with SIMD enabled without avx512 support

**Describe the bug**
My cpu doesn't support avx512
```
[sin3point14@lolikawa build]$ lscpu | grep -o 'avx[^ ]*'
avx
avx2
avx_vnni
```
and the repo doesn't build with SIMD enabled
This results from a faulty cmake check for which I actually already have a fix for this at https://github.com/Continuous-Collision-Detection/Tight-Inclusion/pull/6 . However I saw that this bug is in several dependencies and the contributors are common in the repos, so I decided to open an issue in the most active repo of them all :).

**To Reproduce**
```
$ mkdir build && cd build && cmake ..
$ make -j6
...
[ 29%] Building CXX object _deps/tight-inclusion-build/CMakeFiles/tight_inclusion.dir/src/tight_inclusion/avx.cpp.o
[ 29%] Performing update step for 'ipc_toolkit_test_data_download'
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
[ 30%] Building CXX object _deps/scalable-ccd-build/CMakeFiles/scalable_ccd.dir/src/scalable_ccd/broad_phase/aabb.cpp.o
[ 30%] Built target absl_raw_logging_internal
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
-- Already at requested ref: 333aed25f87c5820bec8a7f82d7959ff87a9a502
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
make[2]: *** [_deps/tight-inclusion-build/CMakeFiles/tight_inclusion.dir/build.make:93: _deps/tight-inclusion-build/CMakeFiles/tight_inclusion.dir/src/tight_inclusion/ccd.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
make[2]: *** [_deps/tight-inclusion-build/CMakeFiles/tight_inclusion.dir/build.make:79: _deps/tight-inclusion-build/CMakeFiles/tight_inclusion.dir/src/tight_inclusion/avx.cpp.o] Error 1
[ 30%] Building CXX object _deps/tight-inclusion-build/CMakeFiles/tight_inclusion.dir/src/tight_inclusion/interval_root_finder.cpp.o
[ 30%] No patch step for 'ipc_toolkit_test_data_download'
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
c++: error: unrecognized command-line option ‘-mno-avx512pf’; did you mean ‘-mno-avx512f’?
c++: error: unrecognized command-line option ‘-mno-avx512er’; did you mean ‘-mno-avx512bw’?
...
```


**Expected behavior**
it should build

**Environment (please complete the following information):**
```
[sin3point14@lolikawa build]$ cat /etc/os-release 
NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo
[sin3point14@lolikawa build]$ uname -r
6.14.6-arch1-1.2-g14
[sin3point14@lolikawa build]$ g++ --version
g++ (GCC) 15.1.1 20250425
Copyright (C) 2025 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[sin3point14@lolikawa build]$ cmake --version
cmake version 4.0.2-dirty

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

--- a/cmake/recipes/tight_inclusion.cmake
+++ b/cmake/recipes/tight_inclusion.cmake
@@ -7,7 +7,7 @@ endif()
 message(STATUS "Third-party: creating target 'tight_inclusion::tight_inclusion'")
 
 include(CPM)
-CPMAddPackage("gh:Continuous-Collision-Detection/Tight-Inclusion@1.0.5")
+CPMAddPackage("gh:Continuous-Collision-Detection/Tight-Inclusion@1.0.6")
 
 # Folder name for IDE
 set_target_properties(tight_inclusion PROPERTIES FOLDER "ThirdParty")

